### ISSUE TITLE
module 'keras.utils' has no attribute 'generic_utils'

### ISSUE THREAD

**Issue by AbtinDjavadifar** [qubvel/segmentation_models#374]
Hi,

I'm trying to run `multiclass segmentation (camvid).ipynb` example and get an error after running the following block:
```
import segmentation_models as sm
```

Here is the error:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-28-93aee672c619> in <module>
----> 1 import segmentation_models as sm
      2 
      3 # segmentation_models could also use `tf.keras` if you do not have Keras installed
      4 # or you could switch to other framework using `sm.set_framework('tf.keras')`

E:\PycharmProjects\Autometrics\Spattering_segmentation\segmentation_models\__init__.py in <module>
----> 1 from .segmentation_models import *

E:\PycharmProjects\Autometrics\Spattering_segmentation\segmentation_models\segmentation_models\__init__.py in <module>
     96 _framework = os.environ.get('SM_FRAMEWORK', _DEFAULT_KERAS_FRAMEWORK)
     97 try:
---> 98     set_framework(_framework)
     99 except ImportError:
    100     other = _TF_KERAS_FRAMEWORK_NAME if _framework == _KERAS_FRAMEWORK_NAME else _KERAS_FRAMEWORK_NAME

E:\PycharmProjects\Autometrics\Spattering_segmentation\segmentation_models\segmentation_models\__init__.py in set_framework(name)
     66     if name == _KERAS_FRAMEWORK_NAME:
     67         import keras
---> 68         import efficientnet.keras  # init custom objects
     69     elif name == _TF_KERAS_FRAMEWORK_NAME:
     70         from tensorflow import keras

~\AppData\Roaming\Python\Python37\site-packages\efficientnet\keras.py in <module>
     15 preprocess_input = inject_keras_modules(model.preprocess_input)
     16 
---> 17 init_keras_custom_objects()

~\AppData\Roaming\Python\Python37\site-packages\efficientnet\__init__.py in init_keras_custom_objects()
     69     }
     70 
---> 71     keras.utils.generic_utils.get_custom_objects().update(custom_objects)
     72 
     73 

AttributeError: module 'keras.utils' has no attribute 'generic_utils'
```

Can anyone help me to solve this?

**Comment by JordanMakesMaps** [comment]
What version of Keras do you have installed? Is it the same as the version required by the version of Segmentation Models you currently have installed?

**Comment by visual-engines** [comment]
same here

```
  File "/home/nvidia/pellame-g4f/lam/backend/lambackend/services/analysis_svc.py", line 6, in <module>
    import segmentation_models as sm
  File "/home/nvidia/virtualenvs/venv_lam/lib/python3.6/site-packages/segmentation_models/__init__.py", line 98, in <module>
    set_framework(_framework)
  File "/home/nvidia/virtualenvs/venv_lam/lib/python3.6/site-packages/segmentation_models/__init__.py", line 68, in set_framework
    import efficientnet.keras  # init custom objects
  File "/home/nvidia/virtualenvs/venv_lam/lib/python3.6/site-packages/efficientnet/keras.py", line 17, in <module>
    init_keras_custom_objects()
  File "/home/nvidia/virtualenvs/venv_lam/lib/python3.6/site-packages/efficientnet/__init__.py", line 71, in init_keras_custom_objects
    keras.utils.generic_utils.get_custom_objects().update(custom_objects)
AttributeError: module 'keras.utils' has no attribute 'generic_utils'
```

Installed versions (inside a v.env. on Jetson TX2):
Keras                  2.4.3
Keras-Applications     1.0.8
segmentation-models    1.0.1
tensorflow             2.2.0+nv20.6

**Comment by AbtinDjavadifar** [comment]
> What version of Keras do you have installed? Is it the same as the version required by the version of Segmentation Models you currently have installed?

Yes, it's the same.

**Comment by AbtinDjavadifar** [comment]
> same here
> 
> ```
>   File "/home/nvidia/pellame-g4f/lam/backend/lambackend/services/analysis_svc.py", line 6, in <module>
>     import segmentation_models as sm
>   File "/home/nvidia/virtualenvs/venv_lam/lib/python3.6/site-packages/segmentation_models/__init__.py", line 98, in <module>
>     set_framework(_framework)
>   File "/home/nvidia/virtualenvs/venv_lam/lib/python3.6/site-packages/segmentation_models/__init__.py", line 68, in set_framework
>     import efficientnet.keras  # init custom objects
>   File "/home/nvidia/virtualenvs/venv_lam/lib/python3.6/site-packages/efficientnet/keras.py", line 17, in <module>
>     init_keras_custom_objects()
>   File "/home/nvidia/virtualenvs/venv_lam/lib/python3.6/site-packages/efficientnet/__init__.py", line 71, in init_keras_custom_objects
>     keras.utils.generic_utils.get_custom_objects().update(custom_objects)
> AttributeError: module 'keras.utils' has no attribute 'generic_utils'
> ```
> 
> Installed versions (inside a v.env. on Jetson TX2):
> 
> * Keras                  2.4.3
> * Keras-Applications     1.0.8
> * segmentation-models    1.0.1
> * tensorflow             2.2.0+nv20.6

I changed `keras.utils.generic_utils.get_custom_objects().update(custom_objects)` to `keras.utils.get_custom_objects().update(custom_objects)` in `.../lib/python3.6/site-packages/efficientnet/__init__.py` and it solved the issue.

**Comment by Zhanglibingo** [comment]
you can install tensorflow-gpu by conda install tensorflow-gpu==1.14.0 again.

**Comment by 4g** [comment]
*TLDR *: export SM_FRAMEWORK=tf.keras before running your code.

error happens in efficientnet.keras package but not in
efficientnet.tfkeras. But importing segmentation_models initializes keras
backend by default as per (Line 48, segmentation_models/__init__.py):

def framework():
    """Return name of Segmentation Models framework"""

return _KERAS_FRAMEWORK

So, sm.set_framework cannot be called before this error happens. You can
either modify the __init__.py to return _TF_KERAS_FRAMEWORK_NAME or export
it in command line as follows.
export SM_FRAMEWORK=tf.keras before running your code.

**Comment by vikalpjain91** [comment]
@qubvel  Can someone have a look? We are unable to reuse your code. (Practically we can via forking your code.) But it is better if we can fix it in stable branch.

**Comment by m1Vahid** [comment]
Hi, I used segmentation-models for building segmentation in google Colab . 
my code worked correctly until yesterday. But today I have this error: 
AttributeError: module 'keras.utils' has no attribute 'generic_utils'
How can i fix it?

**Comment by vikalpjain91** [comment]
@JordanMakesMaps  any update? Can you please provide workaround? I believed requirements.txt should be correct atleast to make it work!

**Comment by vikalpjain91** [comment]
Hi all this is the correct requirements to make it working! working after multiple trial and errors.
!pip install keras==2.3.1
!pip install tensorflow==2.1.0
!pip install keras_applications==1.0.8
!pip install image-classifiers==1.0.0
!pip install efficientnet==1.0.0

**Comment by vikalpjain91** [comment]
Same Issue:
https://github.com/qubvel/segmentation_models/issues/380

**Comment by AbtinDjavadifar** [comment]
> Hi all this is the correct requirements to make it working! working after multiple trial and errors.
> !pip install keras==2.3.1
> !pip install tensorflow==2.1.0
> !pip install keras_applications==1.0.8
> !pip install image-classifiers==1.0.0
> !pip install efficientnet==1.0.0

Thanks. Could you start training using tf.keras instead of keras? I really need to do that as I want to convert the trained model to tf lite and the training has to be done via tf.keras.

**Comment by sezazqureshi** [comment]
%env SM_FRAMEWORK=tf.keras

try this before importing segmentation models , it solved my problem.

**Comment by davidkwast** [comment]
tensorflow = "<2.3"
keras = "<2.4"
albumentations = "==0.4.6"
segmentation-models = "==1.0.1"

**Comment by namanbansalcodes** [comment]
> %env SM_FRAMEWORK=tf.keras
> 
> try this before importing segmentation models , it solved my problem.

This solved it for me on Google Colab.
You are a life saver.

**Comment by tarunnith** [comment]
downgrade the tensorflow to 2.1.0

pip install tensorflow==2.1.0

**Comment by kool7** [comment]
> %env SM_FRAMEWORK=tf.keras
> 
> try this before importing segmentation models , it solved my problem.

Solved my problem on Kaggle thank you.

**Comment by JordanMakesMaps** [comment]
> Hi all this is the correct requirements to make it working! working after multiple trial and errors.
> !pip install keras==2.3.1
> !pip install tensorflow==2.1.0
> !pip install keras_applications==1.0.8
> !pip install image-classifiers==1.0.0
> !pip install efficientnet==1.0.0

Thanks @vikalpjain91. Uninstalling everything, then downloading/installing in this order: 
tensorflow
keras
segmentation-models

worked for me.

**Comment by harshwalia36** [comment]
> %env SM_FRAMEWORK=tf.keras
> 
> try this before importing segmentation models , it solved my problem.

IT worked for me

**Comment by jiajia0818-fighting** [comment]
> %env SM_FRAMEWORK=tf.keras
> 
> try this before importing segmentation models , it solved my problem.

It works.Thanks a lot!

**Comment by OmarSalah26** [comment]
> %env SM_FRAMEWORK=tf.keras
> 
> try this before importing segmentation models , it solved my problem.

big thanks this work to me

**Comment by kaushik25T** [comment]
> Hi all this is the correct requirements to make it working! working after multiple trial and errors.
> !pip install keras==2.3.1
> !pip install tensorflow==2.1.0
> !pip install keras_applications==1.0.8
> !pip install image-classifiers==1.0.0
> !pip install efficientnet==1.0.0

Thanks dude

**Comment by nikhilwohlig** [comment]
> %env SM_FRAMEWORK=tf.keras
> 
> try this before importing segmentation models , it solved my problem.

Worked for me ! Thanks

**Comment by wndxwilson** [comment]
> > %env SM_FRAMEWORK=tf.keras
> > try this before importing segmentation models , it solved my problem.
> 
> Worked for me ! Thanks

this works

**Comment by markusuwe** [comment]
I ran into the same problem (using tensorflow 2.3.2) and I can't downgrade TF due to other dependencies. Is there any chance that this codebase is getting an update so that the more current versions of TF can be used with it?

**Comment by JoshLikesBeer** [comment]
> I ran into the same problem (using tensorflow 2.3.2) and I can't downgrade TF due to other dependencies. Is there any chance that this codebase is getting an update so that the more current versions of TF can be used with it?

Replacing all instances of 
`import keras.whatever`
with
`import tensorflow.keras.whatever`

Fixes the issue for me in tensorflow 2.4.0 and 2.5.0-rc1.

**Comment by qubvel** [comment]
Please create a PR if you cone up with solution

**Comment by shubham10divakar** [comment]
For kaggle installing in this order solved my issue:-
!pip install tensorflow==2.1.0
!pip install keras==2.3.1
!pip install keras_applications==1.0.8
!pip install image-classifiers==1.0.0
!pip install efficientnet==1.0.0
!pip install -U segmentation-models==1.0.1

Link to kaggle notebook - https://www.kaggle.com/shubhamdivakar/qubvel-segmentation-code/

**Comment by P66094125** [comment]
I have the same issue, but i try above of them.
!pip install tensorflow==2.1.0
!pip install keras==2.3.1
!pip install keras_applications==1.0.8
!pip install image-classifiers==1.0.0
!pip install efficientnet==1.0.0
!pip install -U segmentation-models==1.0.1

%env SM_FRAMEWORK=tf.keras

But I try it on jupyter notebook still has the problem.
can anyone help me to solve it?

**Comment by tarunnith** [comment]
> I have the same issue, but i try above of them.
> !pip install tensorflow==2.1.0
> !pip install keras==2.3.1
> !pip install keras_applications==1.0.8
> !pip install image-classifiers==1.0.0
> !pip install efficientnet==1.0.0
> !pip install -U segmentation-models==1.0.1
> 
> %env SM_FRAMEWORK=tf.keras
> 
> But I try it on jupyter notebook still has the problem.
> Is there has any method to fix it?

I have not worked on the Jupyter notebook but for the kaggle   %env SM_FRAMEWORK=tf.keras   woks fine.

**Comment by zahidmb** [comment]
> 
> 
> > I ran into the same problem (using tensorflow 2.3.2) and I can't downgrade TF due to other dependencies. Is there any chance that this codebase is getting an update so that the more current versions of TF can be used with it?
> 
> Replacing all instances of
> `import keras.whatever`
> with
> `import tensorflow.keras.whatever`
> 
> Fixes the issue for me in tensorflow 2.4.0 and 2.5.0-rc1.

While converting from keras to tensorflow.keras, I am facing some issues like cannot convert following;

from keras.engine.topology import Network

Can you help me how to convert it to tf.keras. Tq.

**Comment by francesco-taioli** [comment]
```
import os
os.environ["SM_FRAMEWORK"] = "tf.keras" #before the import
from segmentation_models import PSPNet
```
Solved the issue for me

**Comment by creativesalam** [comment]
Using TensorFlow 2.5, no problem with after following. I am using Keras from tensorflow. 

**Option 1:**  In the start of notebook, change the environment variable as 
`%env SM_FRAMEWORK=tf.keras`

**Option 2:**  Another way to change environment variable. 
`import segmentation_models as sm
sm.set_framework('tf.keras')`

Problem solved for TensorFlow 2.5.

**Comment by salimsoltani28** [comment]
> > %env SM_FRAMEWORK=tf.keras
> > try this before importing segmentation models , it solved my problem.
> 
> This solved it for me on Google Colab. You are a life saver.

Worked for me as well!

**Comment by Jordan-Pierce** [comment]
> > %env SM_FRAMEWORK=tf.keras
> > try this before importing segmentation models , it solved my problem.
> 
> This solved it for me on Google Colab. You are a life saver.

This also solved it for me, thanks!

**Comment by Deewourne** [comment]
> %env SM_FRAMEWORK=tf.keras
> 
> try this before importing segmentation models , it solved my problem.

It also worked for me, could you please explain what this does?

**Comment by Masumekeshavarzi** [comment]
It refers to changes in new version of keras while segmentation model is built with older versions of keras
In Linux I ran these codes : 
1. cat _(error path)_
2. sed -i 's/keras.utils.generic_utils.get_custom_objects()/keras.utils.get_custom_objects()/g' _error path_

**Comment by keepfighting666** [comment]
> %env SM_FRAMEWORK=tf.keras
> 
> try this before importing segmentation models , it solved my problem.

it works.

**Comment by myyyy1987** [comment]
> %env SM_FRAMEWORK=tf.keras
> 
> try this before importing segmentation models , it solved my problem.

Thanks! it works for me.
